### PR TITLE
8307117: TextArea: wrapText property ignored when changing font

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
Requesting content layout when font changes.

This change makes the visual impact of [JDK-8314683](https://bugs.openjdk.org/browse/JDK-8314683) more visible, so perhaps both bugs should be fixed at the same time.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8307117](https://bugs.openjdk.org/browse/JDK-8307117): TextArea: wrapText property ignored when changing font (**Bug** - P4)

### Reviewers
 * [Karthik P K](https://openjdk.org/census#kpk) (@karthikpandelu - Committer)
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1217/head:pull/1217` \
`$ git checkout pull/1217`

Update a local copy of the PR: \
`$ git checkout pull/1217` \
`$ git pull https://git.openjdk.org/jfx.git pull/1217/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1217`

View PR using the GUI difftool: \
`$ git pr show -t 1217`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1217.diff">https://git.openjdk.org/jfx/pull/1217.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1217#issuecomment-1930854030)

### CODE REVIEW

**Comment by andy-goryachev-oracle** [comment]
keep alive

**Comment by andy-goryachev-oracle** [comment]
keep alive

**Comment by andy-goryachev-oracle** [comment]
on second thought, let's address this independently of [JDK-8314683](https://bugs.openjdk.org/browse/JDK-8314683).

**Comment by kevinrushforth** [comment]
Reviewers: @arapte @karthikpandelu 

This seems reasonable, but I'll let Ambarish and Karthik review. My only question is whether it would it be possible to create an automated test for this?

File: modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextAreaSkin.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2011, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (arapte) [inline modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextAreaSkin.java:2]: minor: year should be 2024.
We update copyright years in bulk before releases/new year, so in general one can avoid changing copyright year along with fix.

**Comment by karthikpandelu** [comment]
I can also see the same behaviour which @arapte mentioned above.

**Comment by karthikpandelu** [comment]
The flickering looks to be gone now but I see another issue. When the font size is changed from larger value like 144 to a smaller value like 10, the position of text in the TextArea shifts.
Screenshot 2024-02-08 at 3 35 50 PM
When monkey tester window loses focus, it goes back to the original position.

Steps to reproduce the issue:
1. Open the MonkeyTester with TextArea having larger font size like 144. (If smaller font size is present, change the font size without clicking on the text)
2. Click on the text to get caret.
3. Change font size to 10.
4. Click on the text to get caret again, the text shifts.

Not sure if this is related to the other issues you have mentioned in this PR before.

**Comment by andy-goryachev-oracle** [comment]
> the position of text in the TextArea shifts.

yes, that's [JDK-8314683](https://bugs.openjdk.org/browse/JDK-8314683)
the padding size is expressed in ems which is incorrect - it should be fixed like everywhere else.

**Comment by andy-goryachev-oracle** [comment]
> would it be possible to create an automated test for this?

created a robot test.

**Review by karthikpandelu — APPROVED** [review]
LGTM

**Review by arapte — APPROVED** [review]

**Pull request closed without merging**